src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -0,0 +1,165 @@
+            {
+                if (map.TryGetValue(param.Name, out var paramValue))
+                {
+                    var valueStr = paramValue.ToObject().ToString();
+
+                    switch (param.In.Value)
+                    {
+                        case ParameterLocation.Path:
+                            path = path.Replace("{" + param.Name + "}", HttpUtility.UrlEncode(valueStr));
+                            break;
+
+                        case ParameterLocation.Query:
+                            query.Append((query.Length == 0) ? "?" : "&");
+                            query.Append(param.Name);
+                            query.Append('=');
+                            query.Append(HttpUtility.UrlEncode(valueStr));
+                            break;
+
+                        default:
+                            throw new NotImplementedException($"{param.In}");
+                    }
+                }
+            }
+
+            var url = path + query.ToString();
+
+            // $$$ Not handling Body yet...
+            var request = new HttpRequestMessage(_method, url);
+            return request;
+        }   
+
+        public async Task<FormulaValue> DecodeResponseAsync(HttpResponseMessage response)
+        {
+            var json = await response.Content.ReadAsStringAsync();
+
+            if (!response.IsSuccessStatusCode)
+            {
+                var msg = $"Connector called failed {response.StatusCode}): " + json;
+
+                // $$$ Do any connectors have 40x behavior here in their response code?
+                // or 201 long-ops behavior?
+
+                // $$$ Still type this. 
+                return FormulaValue.NewError(new ExpressionError
+                {
+                    Kind = ErrorKind.Unknown,
+                    Message = msg
+                });
+            }
+
+            if (string.IsNullOrWhiteSpace(json))
+            {
+                return FormulaValue.NewBlank(_returnType);
+            }
+
+            // $$$ Proper marshalling?,  use _returnType;
+            // If schema was an array, we returned a Single Column Table type for it. 
+            // Need to ensure we marshal it consistency here. 
+            var result = FormulaValue.FromJson(json);            
+
+            return result;
+        }
+
+        public async Task<FormulaValue> InvokeAsync(string cacheScope, CancellationToken cancel, FormulaValue[] args)
+        {
+            FormulaValue result;
+            var request = BuildRequest(args);
+
+            var key = request.RequestUri.ToString();
+
+            if (request.Method != HttpMethod.Get)
+            {
+                _cache.Reset(cacheScope);
+                key = null; // don't bother caching
+            }
+                        
+            var result2 = await _cache.TryGetAsync(cacheScope, key, async () =>
+            {
+                var response = await _httpClient.SendAsync(request, cancel);
```

Comment:
Should we add .ConfigureAwait(false) ?

Reply:
Comment applies to all async calls

Reply:
Is it necessary? Else we have 1000+ awaits to go update.

Reply:
If we want to avoid deadlocks when loaded in asp.net ... yes